cypress/integration/sample_spec.js

```diff
@@ -80,13 +86,144 @@ describe('popcorn button', () => {
+        // search button should be disabled
+    })
+})
+*/
+
+describe('dropdown list for no. of search results', () => {
+
+    beforeEach(() => {
+        cy.visit('/')
+        cy.get('[data-cy=dropdown-number-results]') 
+            .as ('dropdown');
+    }) 
 
- 
\ No newline at end of file
+    it('renders dropdown element & displays default no. of results', () => {
+        cy.get('@dropdown')
+            .first()
+            .should('contain', numberOfResults); 
+    })
+
+    it('allows user to select a dropdown list option - using > CSS Selector ', () => {
+        cy.get('@dropdown').click() // reveals dropdown which contains ul 
+        cy.get('ul>li') // yields all li items in ul - not sure why it only finds one ul though? 
+            .eq(numberOfResults - 1) // selects option by index. can also do .first()  or .last()
+            .click() 
+    })
+
+    it('updates to display the selected value - using .within', () => {
+        cy.get('@dropdown').click()
+        cy.get('ul') // yields all ul elements, of which the dropdown is the second 
+            .eq(1)
+            .within(($list) => {
+                cy.contains(numberOfResults - 1).click() // or cy.get('li:nth-child(9)').click() 
+            })  
+        cy.contains(numberOfResults - 1); // assertion
+    })
+})
+
+describe('keyword responsive text', () => {
+
+    beforeEach(() => {
+        cy.visit('/');
+    })
+
+    it('renders initial text', () => {
+        cy.contains('The keyword is ???')
+    })
+    
+    it('updates with input field value - test 1', () => {
+        cy.get('[data-cy=input-field]').type('test');
+        cy.contains('The keyword is test');
+    })
+    
+    it('updates with input field value - test 2', () => {
+        cy.get('[data-cy=input-field]').type('test');
+        cy.get('[data-cy=responsive-text]')
+        .invoke('text')
+        .should('contain','The keyword is test');
+    })
+})
+
+describe('results responsive text', () => {
+
+    beforeEach(() => {
+        cy.visit('/');
+    })
+
+    it('renders initial text', () => {
+        cy.contains('Press SEARCH to return 10 results');
+    })
+
+    it.only('updates text with selected dropdown value', () => {
+        cy.get('[data-cy=dropdown-number-results]').click()
+        cy.get('ul>li')
+            .eq(numberOfResults-2) // zero-based index 
+            .click()
+        cy.get('[data-cy=responsive-text]')
+            .invoke('text')
+            .should('contain', `Press SEARCH to return ${numberOfResults-1} results`);
+    })
+})
+
+describe('search function', () => {
+
+    beforeEach(() => {
+        cy.visit('/')
+        // TODO: define aliases for input field & search button
+    })
+    
+    it('accepts a search term and returns search results', () => {
+        cy.get('[data-cy=input-field]').type('the notebook')
+        cy.get('[data-cy=search-button]').click() // alternative: cy.contains('Search').click()
+        cy.wait(500); 
```

Comment:
This will introduce a source of flakiness where the test will fail if the API ever takes >500ms to respond. It would be good to stub out the response if possible - although this is fine for the purposes of learning for now.

Reply:
This would mean to pass it a mock response (i.e mock JSON data), right? 
Alternatively I could look into something like JS await. Cypress seems to have options like cypress-wait-until.
Although I do acknowledge it's a vunerability to incl. API calls in our tests.